src/main/java/com/zendesk/maxwell/Maxwell.java

```diff
@@ -179,6 +182,29 @@ private void start() throws Exception {
 
 		this.context.start();
 		this.onReplicatorStart();
+
+		// Background: Dropwizard throws an exception if you try to register multiple metrics with the same name.
+		// We register everything here for simplicity. In reality we only need to register the replication.lag metric
+		// here.
+		//
+		// More details:
+		// Register the replication lag metric here because there are codepaths that create multiple replicators
+		// (at least in the tests).
+		// In addition, the tests create multiple BufferedMaxwells so we need to avoid those cases too...
+		if ( !(this instanceof BufferedMaxwell) ) {
```

Comment:
hmmm, what if instead of doing this we instead just prevented ourselves from double-registering metrics, something like:

```
  if ( !MaxwellMetrics.isRegistered(MetricRegistry.name(...)) ) { 

  }
```